test/deploy-pipeline.sh

```diff
@@ -47,12 +47,18 @@ cd ${DIR}/${KFAPP}
 
 ## Update pipeline component image
 pushd ks_app
+# Delete pipeline component first before applying so we guarantee the pipeline component is new.
+ks delete default -c pipeline
+sleep 60s
```

Comment:
Curious why add the sleep here? Is the 'ks delete' nonblocking? If so, we might need to wait until the deletion completes here.

Reply:
yes it's async delete. ks apply wont fail fast if it's applying to a terminating resource. 
sleep 60s would ensure things are cleaned up.

Reply:
I think the longer plan should be building the pipeline images before deploying the kubeflow and have the kubeflow deployment include the new images. 
On the other hand, I thought the delete is a blocking call. If it is async, depending on a hardcoded timing is not good practice. If the deletion takes less than 60 seconds, we are wasting time in the tests; If the deletion takes more than 60 seconds, it is not working. Above all, I think we need a more deterministic way to wait for the pipeline deletion. 
WDYT?